src/workspaces/NewWorkspaceModal/NewWorkspaceWizard.tsx

```diff
@@ -0,0 +1,798 @@
+import validate from 'validate.js';
+
+import { AccessEntry, WorkspaceAcl } from '../acl-utils';
+import { AclInput } from '../ShareWorkspaceModal/Collaborator';
+import { CurrentCollaborators } from '../ShareWorkspaceModal/CurrentCollaborators';
+
+const constraints = {
+  name: {
+    presence: { allowEmpty: false },
+    length: { maximum: 254 },
+    format: {
+      pattern: /[\w- ]*/,
+      message: 'can only contain letters, numbers, dashes, underscores, and spaces',
+    },
+  },
+  namespace: {
+    presence: true,
+  },
+};
+
+const invalidBillingAccountMsg =
+  'Workspaces may only be created in billing projects that have a Google billing account accessible in Terra';
+
+const ariaInvalidBillingAccountMsg = (invalidBillingAccount: boolean): string => {
+  return invalidBillingAccount ? ` with warning "${invalidBillingAccountMsg}"` : '';
+};
+
+export interface NewWorkspaceWizardProps {
+  buttonText?: string;
+  cloneWorkspace?: WorkspaceWrapper;
+  cloudPlatform?: CloudPlatform;
+  renderNotice?: (args: { selectedBillingProject?: BillingProject }) => ReactNode;
+  requiredAuthDomain?: string;
+  requireEnhancedBucketLogging?: boolean;
+  title?: string;
+  waitForServices?: {
+    wds?: boolean;
+  };
+  workflowImport?: boolean;
+  onDismiss: () => void;
+  onSuccess: (newWorkspace: WorkspaceInfo) => void;
+}
+
+export const NewWorkspaceWizard = withDisplayName(
+  'NewWorkspaceWizard',
+  ({
+    cloneWorkspace,
+    cloudPlatform,
+    onSuccess,
+    onDismiss,
+    renderNotice = () => null,
+    requiredAuthDomain,
+    requireEnhancedBucketLogging,
+    title,
+    buttonText,
+    workflowImport,
+  }: NewWorkspaceWizardProps) => {
+    // State
+    const [billingProjects, setBillingProjects] = useState<BillingProject[]>();
+    const [allGroups, setAllGroups] = useState<CurrentUserGroupMembership[]>();
+    const [name, setName] = useState(cloneWorkspace ? `${cloneWorkspace.workspace.name} copy` : '');
+    const [namespace, setNamespace] = useState(cloneWorkspace ? cloneWorkspace.workspace.namespace : undefined);
+    const [description, setDescription] = useState(cloneWorkspace?.workspace.attributes?.description || '');
+    const [groups, setGroups] = useState<string[]>([]);
+    const [enhancedBucketLogging, setEnhancedBucketLogging] = useState(!!requireEnhancedBucketLogging);
+    const [nameModified, setNameModified] = useState(false);
+    const [loading, setLoading] = useState(true);
+    const [creating, setCreating] = useState(false);
+    const [createError, setCreateError] = useState<string>();
+    const [bucketLocation, setBucketLocation] = useState(defaultLocation);
+    const [sourceGCPWorkspaceRegion, setSourceGCPWorkspaceRegion] = useState<string>(defaultLocation);
+    const [sourceGCPWorkspaceRegionError, setSourceGCPWorkspaceRegionError] = useState(false);
+    const [isAlphaRegionalityUser, setIsAlphaRegionalityUser] = useState(false);
+    const [activeTab, setActiveTab] = useState<string>('basic');
```

Comment:
Did you think about putting numbers as keys for the tabs? That way if they change or the order changes it will always default to the first tab?

Reply:
But it seems kinda confusing to have a number because then anyone looking doesn't know which tab it is

Reply:
Looking at the code, you mean? I think that would be helped by having defined `Tab` objects, but all my feedback should be considered optional, since it is not often I am in this area of the code in comparison to you all.